app/middleware/postcodeLookup.js

```diff
@@ -0,0 +1,60 @@
+const postcodeValidator = require('../lib/postcodeValidator');
+const log = require('../lib/logger');
+const PostcodesIOClient = require('postcodesio-client');
+
+// rewire (a framework for mocking) doesn't support const
+// eslint-disable-next-line no-var
+var PostcodesIO = new PostcodesIOClient();
+// eslint-disable-next-line no-var
+var renderer = require('./renderer');
+
+function outsideEngland(outcodeDetails) {
+  return !outcodeDetails.country.some(c => c === 'England');
+}
+
+function lookupPostcode(req, res, next) {
+  const postcode = res.locals.postcode;
+
+  if (postcodeValidator.isOutcode(postcode)) {
+    PostcodesIO.outcode(postcode, (err, outcodeDetails) => {
+      if (outcodeDetails && outsideEngland(outcodeDetails)) {
+        log.info('validate-outcode-notEnglish');
+        renderer.postcodeNotEnglish(postcode, req, res);
+        log.info('validate-outcode-notEnglish');
+      } else if (outcodeDetails) {
+        res.locals.location = { lat: outcodeDetails.latitude, lon: outcodeDetails.longitude };
+        res.locals.isOutcode = true;
+        log.info(`outcode ${JSON.stringify(res.locals.location)}`);
```

Comment:
Rather than stringify the object it can be logged via:
`log.info({ postcodeIOResponse: { outcodeDetails } }, 'PostcodeIO outcode response');`
I'd also suggest doing the logging once on line 21 i.e. before any of the ifs. 
Consider whether `info` or `debug` would be best.